applications/flash_attention_v2/kernel/xe_flash_attn_decode.hpp

```diff
@@ -439,13 +439,24 @@ class FMHADecode {
             }
           }
         }
+      }
 
-        CollectiveSoftmaxEpilogue softmax(params.softmax);
-        softmax.template operator()<Num_SGs>((kv_splits - 1) == 0, tSr, max_reg, sum_reg, shmem_max_tensor, out_reg);
-
-        collective_mma.template mmaPV<VSlicer>(out_reg, tSr, gV, out_reg, mainloop_params, false, curr_kv_tile_idx);
+      if (seq_len_kv % QK_BLK_N != 0) {
+        int col_idx = (kv_tile_idx + (kv_splits_new - 1) * ATOM_M) * QK_SG_N + thread_idx % SubgroupSize;
+        int remainder = seq_len_kv % QK_BLK_N;
+        CUTLASS_PRAGMA_UNROLL
+        for (int n = 0; n < FragsN; n++, col_idx += get<1>(MmaAtomShape())) { // 4
+          if (col_idx > remainder) {
+            tSr(0, 0, n) = ElementAccumulator{-INFINITY};
+          }
+        }
```

Comment:
Is anything other than this actual needed to fix the accuracy issue?

If not the other changes should not be part of this PR. And the weird loop structure was on purpose to work around an IGC issue.

Reply:
why does that change require the change in loop bound on line 352?